### PR TITLE
[rtve] Add support for audio only

### PR DESCRIPTION
**Pull request by kebianizao**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [ ] New extractor
- [X] New feature

---

### Description of your *pull request* and other information

This adds support to audio only urls to the rtve extractor. So far this extractor did not support urls like Audio only episode . With this extension such audio only url is supported.

The class supporting the above mentioned url includes a test case which runs successfully here.

Taken over from https://github.com/ytdl-org/youtube-dl/pull/29023

### CODE REVIEW

File: yt_dlp/extractor/extractors.py [hunk 1]
```diff
@@ -1230,7 +1230,13 @@
...
 from .rtp import RTPIE
 from .rts import RTSIE
-from .rtve import RTVEALaCartaIE, RTVELiveIE, RTVEInfantilIE, RTVELiveIE, RTVETelevisionIE
+from .rtve import (
+        RTVEALaCartaIE,
+        RTVELiveIE,
+        RTVEAudioIE,
+        RTVEInfantilIE,
+        RTVETelevisionIE,
+)
```
Comment (pukkandan) [inline yt_dlp/extractor/extractors.py:1239]: use 4 spaces and sort alphabetically

File: yt_dlp/extractor/rtve.py [hunk 2]
```diff
@@ -178,6 +178,87 @@ def _get_subtitles(self, video_id, sub_file):
...
+            audio_id)['page']['items'][0]
+        title = info['title'].strip()
+        formats = self._extract_png_formats(audio_id)
+
+        return {
+            'id': audio_id,
+            'title': title,
+            'thumbnail': info.get('thumbnail'),
+            'duration': float_or_none(info.get('duration'), 1000),
+            'series': info.get('programInfo').get('title'),
```
Comment (pukkandan) [inline yt_dlp/extractor/rtve.py:257]:
```suggestion
            'series': try_get(info, lambda x: x['programInfo']['title']),
```
or 
```py
            'series': traverse_obj(info, ('programInfo', 'title')),
```

File: yt_dlp/extractor/rtve.py [hunk 3]
```diff
@@ -178,6 +178,87 @@ def _get_subtitles(self, video_id, sub_file):
...
+            'id': '6082623',
+            'ext': 'mp3',
+            'title': 'Capítulo 26 y último: La muerte de Victor',
+            'thumbnail': r're:https?://.+/1632147445707.jpg',
+            'duration': 3174.086,
+        },
+    }]
+
+    def _extract_png_formats(self, audio_id):
+        png = self._download_webpage(
```
Comment (pukkandan) [inline yt_dlp/extractor/rtve.py:219]: what's png?

File: yt_dlp/extractor/rtve.py [hunk 4]
```diff
@@ -178,6 +178,87 @@ def _get_subtitles(self, video_id, sub_file):
...
+            'http://www.rtve.es/ztnr/movil/thumbnail/%s/audios/%s.png' %
+            (self._manager, audio_id),
+            audio_id, 'Downloading url information', query={'q': 'v2'})
+        q = qualities(['Media', 'Alta', 'HQ', 'HD_READY', 'HD_FULL'])
+        formats = []
+        for quality, audio_url in self._decrypt_url(png):
+            ext = determine_ext(audio_url)
+            print(quality, audio_url, ext)
```
Comment (pukkandan) [inline yt_dlp/extractor/rtve.py:227]: remove

File: yt_dlp/extractor/rtve.py [hunk 5]
```diff
@@ -178,6 +178,87 @@ def _get_subtitles(self, video_id, sub_file):
...
+        formats = self._extract_png_formats(audio_id)
+
+        return {
+            'id': audio_id,
+            'title': title,
+            'thumbnail': info.get('thumbnail'),
+            'duration': float_or_none(info.get('duration'), 1000),
+            'series': info.get('programInfo').get('title'),
+            'formats': formats,
+        }
```
Comment (pukkandan) [inline yt_dlp/extractor/rtve.py:259]:
```suggestion
        return {
            'id': audio_id,
            'title': info['title'].strip(),
            'thumbnail': info.get('thumbnail'),
            'duration': float_or_none(info.get('duration'), 1000),
            'series': info.get('programInfo').get('title'),
            'formats': self._extract_png_formats(audio_id),
        }
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/rtve.py (hunk 3 above)
Comment (kebianizao) [reply yt_dlp/extractor/rtve.py:219]: This is png thumbnail file which stealthly encodes media details as url or media quality.

File: yt_dlp/extractor/rtve.py (hunk 4 above)
Comment (kebianizao) [reply yt_dlp/extractor/rtve.py:227]: I guess you mean the print. This was slipped by mistake, sorry.

**Comment by kebianizao** [comment]
Thanks for the review. I think I've addressed all comments, but let me know if you see anything else.

File: yt_dlp/extractor/rtve.py (hunk 3 above)
Comment (pukkandan) [reply yt_dlp/extractor/rtve.py:220]: might wanna put that in a comment. It's not obvious from reading the code

File: yt_dlp/extractor/rtve.py [hunk 6]
```diff
@@ -178,6 +179,90 @@ def _get_subtitles(self, video_id, sub_file):
             for s in subs)
 
 
+class RTVEAudioIE(RTVEALaCartaIE):
+    IE_NAME = 'rtve.es:audio'
+    IE_DESC = 'RTVE audio'
+    _VALID_URL = r'https?://(?:www\.)?rtve\.es/(alacarta|play)/audios/[^/]+/[^/]+/(?P<id>[0-9]+)/'
```
Comment (pukkandan) [inline yt_dlp/extractor/rtve.py:185]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?rtve\.es/(alacarta|play)/audios/[^/]+/[^/]+/(?P<id>[0-9]+)'
```

File: yt_dlp/extractor/rtve.py [hunk 7]
```diff
@@ -178,6 +179,90 @@ def _get_subtitles(self, video_id, sub_file):
...
+        info = self._download_json(
+            'https://www.rtve.es/api/audios/%s.json' % audio_id,
+            audio_id)['page']['items'][0]
+
+        return {
+            'id': audio_id,
+            'title': info['title'].strip(),
+            'thumbnail': info.get('thumbnail'),
+            'duration': float_or_none(info.get('duration'), 1000),
+            'series': try_get(info, lambda x: x['programInfo']['title']),
```
Comment (pukkandan) [inline yt_dlp/extractor/rtve.py:261]: none of the tests cover `series`

**Review by pukkandan — APPROVED** [review]
everything else looks good

File: yt_dlp/extractor/rtve.py (hunk 7 above)
Comment (pukkandan) [reply yt_dlp/extractor/rtve.py:261]: @kebianizao Please add a test for series. That's the only thing left for this to be merged

Comment (kebianizao) [reply yt_dlp/extractor/rtve.py:261]: Sorry, I somehow missed your point. I'll take a look as soon as I can.

**Pull request merged**